internal/expr/scan.go

```diff
@@ -148,28 +160,54 @@ func (re *ResultExpr) decodeValue(v reflect.Value) error {
 	return nil
 }
 
-func setValue(dest reflect.Value, fInfo field, val any) error {
+func setValue(dest reflect.Value, fInfo fielder, val any) error {
 	var isZero bool
 
 	v := reflect.ValueOf(val)
+	name := fInfo.Name()
+
+	fmt.Printf("%s, %v\n", name, dest.Type().Kind())
 
-	if val == nil {
-		if fInfo.omitEmpty {
-			return nil
+	switch dest.Type().Kind() {
+	case reflect.Struct:
+		f, ok := fInfo.(field)
+		if !ok {
+			return fmt.Errorf("internal error: argument of type %#v has no member of type %#v", dest.Type(), f)
 		}
-		isZero = true
-		v = reflect.Zero(fInfo.typ)
-	}
 
-	if !isZero && v.Type() != fInfo.typ {
-		return fmt.Errorf("result of type %#v but field %#v is type %#v", v.Type().Name(), fInfo.name, fInfo.typ.Name())
-	}
-	f := dest.FieldByIndex(fInfo.index)
-	if !f.CanSet() {
-		return fmt.Errorf("cannot set field %#v. CanAddr=%v", fInfo.name, f.CanAddr())
+		if val == nil {
+			if f.omitEmpty {
+				return nil
+			}
+			isZero = true
+			v = reflect.Zero(f.typ)
+		}
+		if !isZero && v.Type() != f.typ {
+			return fmt.Errorf("result of type %#v but field %#v is type %#v", v.Type().Name(), name, f.typ.Name())
+		}
+		itsField := dest.FieldByIndex(f.index)
+		if !itsField.CanSet() {
+			return fmt.Errorf("cannot set field %#v. CanAddr=%v", f.name, itsField.CanAddr())
+		}
+		itsField.Set(v)
+		return nil
+	case reflect.Map:
+		m, ok := fInfo.(mapKey)
+		if !ok {
+			return fmt.Errorf("internal error: argument of type %#v has no member of type %#v", dest.Type(), m)
+		}
+
+		fmt.Printf("%s\n", m.name)
+		if val != nil {
+			fmt.Printf("%s\n", m.name)
+			itsKey := dest.MapIndex(reflect.ValueOf(m.name))
+			itsKey.Set(v)
+		}
+		// todo: what would be the zero value of a map value whose type we don't know ?
```

Comment:
It would be `nil` of type `any`

Reply:
Done! Though this may need further testing.